drivers/src/tla2528.cpp

```diff
@@ -0,0 +1,137 @@
+#include <libhal-util/bit.hpp>
+#include <libhal-util/i2c.hpp>
+#include <libhal/error.hpp>
+#include <libhal/units.hpp>
+#include <tla2528.hpp>
+
+namespace sjsu::drivers {
+
+tla2528::tla2528(hal::i2c& p_i2c, hal::byte p_i2c_address)
+  : m_bus(p_i2c)
+{
+  m_i2c_address = p_i2c_address;
+  // TODO: reset command
+}
+
+void tla2528::set_analog_channel(hal::byte p_channel)
+{
+  if (p_channel == m_channel)
+    return;
+  if (p_channel > 7)
+    throw hal::argument_out_of_domain(this);
```

Comment:
Rather than spelling out this check, you can add a function called `throw_if_invalid_channel(p_channel);`.